### PR TITLE
fix crud author, publisher + export excel

### PR DESCRIPTION
**Pull request by hoangtv2605**
## Related Tickets:
- [26959](https://edu-redmine.sun-asterisk.vn/issues/26959)
- [26956](https://edu-redmine.sun-asterisk.vn/issues/26956)

## WHAT this PR does?
- Export list authors, publishers ra file excel
- Fix CRUD authors, publishers (chuyển sang dùng modal bootstrap) 

## Notes Impacted Areas:
- N/A.

## Attachments:
- N/A.

### CODE REVIEW

**Comment by hoangtv2605** [comment]
ready

File: app/Exports/AuthorsExport.php [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+use App\Models\Author;
+use Illuminate\Contracts\View\View;
+use Maatwebsite\Excel\Concerns\FromView;
+
+class AuthorsExport implements FromView
+{
+    public function view(): View
+    {
+        return view('exports.authors', [
+            'authors' => Author::all()
```
Comment (thaonp104) [inline app/Exports/AuthorsExport.php]:
```suggestion
            'authors' => Author::all(),
```

File: app/Exports/PublishersExport.php [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+use App\Models\Publisher;
+use Illuminate\Contracts\View\View;
+use Maatwebsite\Excel\Concerns\FromView;
+
+class PublishersExport implements FromView
+{
+    public function view(): View
+    {
+        return view('exports.publishers', [
+            'publishers' => Publisher::all()
```
Comment (thaonp104) [inline app/Exports/PublishersExport.php]: tương tự

File: resources/css/error.css [hunk 3]
```diff
@@ -15,3 +15,46 @@
...
+  
+.border-none {
+    border: none;
+}
+  
+.max-width {
+    width: 100% !important;
+}
+  
+input[type="email"],input[type="password"]{
```
Comment (thaonp104) [inline resources/css/error.css]:
```suggestion
input[type="email"],input[type="password"] {
```

File: resources/views/admin/publisher/index.blade.php [hunk 4]
```diff
@@ -18,72 +18,75 @@
...
+                                    <th class="text-center">{{ trans('message.email') }}</th>
+                                    <th class="text-center">{{ trans('message.address') }}</th>
+                                    <th class="text-center">{{ trans('request.edit') }}</th>
+                                    <th class="text-center">{{ trans('request.delete') }}</th>
+                                </tr>
+                            </thead>
+                            <tbody>
+                                @foreach ($publishers as $key => $publisher)
+                                    <tr>
+                                        <td class="text-center">{{ $key + 1 }}</td>
```
Comment (thaonp104) [inline resources/views/admin/publisher/index.blade.php]: cho 1 vào config đi nhé

File: resources/views/exports/authors.blade.php [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+            <th>{{ trans('message.description') }}</th>
+            <th>{{ trans('message.email') }}</th>
+            <th>{{ trans('message.created_at') }}</th>
+            <th>{{ trans('message.updated_at') }}</th>
+        </tr>
+    </thead>
+    <tbody>
+        @foreach($authors as $key => $author)
+            <tr>
+                <td>{{ $key + 1 }}</td>
```
Comment (thaonp104) [inline resources/views/exports/authors.blade.php]: tương tự

File: resources/views/exports/publishers.blade.php [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+            <th>{{ trans('message.email') }}</th>
+            <th>{{ trans('message.address') }}</th>
+            <th>{{ trans('message.created_at') }}</th>
+            <th>{{ trans('message.updated_at') }}</th>
+        </tr>
+    </thead>
+    <tbody>
+        @foreach($publishers as $key => $publisher)
+            <tr>
+                <td>{{ $key + 1 }}</td>
```
Comment (thaonp104) [inline resources/views/exports/publishers.blade.php]: tương tự

File: resources/views/exports/authors.blade.php [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+            <th>{{ trans('request.id') }}</th>
+            <th>{{ trans('message.name') }}</th>
+            <th>{{ trans('message.description') }}</th>
+            <th>{{ trans('message.email') }}</th>
+            <th>{{ trans('message.created_at') }}</th>
+            <th>{{ trans('message.updated_at') }}</th>
+        </tr>
+    </thead>
+    <tbody>
+        @foreach($authors as $key => $author)
```
Comment (huyenttm-0901) [inline resources/views/exports/authors.blade.php]: convention

File: resources/views/exports/publishers.blade.php [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+            <th>{{ trans('request.id') }}</th>
+            <th>{{ trans('message.name') }}</th>
+            <th>{{ trans('message.email') }}</th>
+            <th>{{ trans('message.address') }}</th>
+            <th>{{ trans('message.created_at') }}</th>
+            <th>{{ trans('message.updated_at') }}</th>
+        </tr>
+    </thead>
+    <tbody>
+        @foreach($publishers as $key => $publisher)
```
Comment (huyenttm-0901) [inline resources/views/exports/publishers.blade.php]: convention

**Comment by hoangtv2605** [comment]
ready

File: resources/js/show_edit.js [hunk 9]
```diff
@@ -0,0 +1,77 @@
+$(document).ready(function() {
+    $("#listUsers").DataTable({
+        "columnDefs": [
+            { "width": "20px", "targets": 0 },
+            { "width": "100px", "targets": 1 },
+            { "width": "100px", "targets": 2 },
+            { "width": "100px", "targets": 3 },
+            { "width": "70px", "targets": 4, "orderable": false },
+            { "width": "70px", "targets": 5, "orderable": false }
```
Comment (peternguyen287) [inline resources/js/show_edit.js]: { "width": "70px", "targets": 5, "orderable": false },

**Review by huyenttm-0901 — COMMENTED** [review]
conflict

**Comment by hoangtv2605** [comment]
ready

**Pull request merged**